torch/distributed/_functional_collectives.py

```diff
@@ -507,6 +619,30 @@ def cast_listint(x):
 
     return (tag, rankset, group_size)
 
+
+def _expand_group_for_for_native_ops(group: RANK_TYPES, tag: str = "") -> Tuple[str, int]:
+    """
+    Given group in RANK_TYPES, return group name and group size.
+    """
+    # We decided to replace (tag, ranks, group_size) in the previous  with
+    # group_name in functional collective signatures as we migrate to native
+    # c10d_functional. Fow now, we'll raise an error if tag is supplied when
+    # USE_NATIVE_C10D_FUNCTIONAL is set. Once we completely switch to native
+    # c10d_functional, we'll remove `tag` from the Python APIs.
+    #
+    # https://github.com/pytorch/pytorch/pull/109738#discussion_r1334538196
+    if tag != "":
+        warnings.warn(f"Tag \"{tag}\" is ignored since USE_NATIVE_C10D_FUNCTIONAL is set")
+
+    if isinstance(group, dist.ProcessGroup):
+        return group.group_name(), dist.get_world_size(group)
+    elif isinstance(group, str):
+        return group, torch._C._distributed_c10d._resolve_process_group(group).size()
+    else:
+        # TODO(yifu): DeviceMesh supported will be added in a subsequent PR
```

Comment:
I think device mesh support is actually quite simple to add?
i.e. we can do `device_mesh.get_dim_groups()` and then we will get the process group, then we can handle it the same way as `ProcessGroup`, the complexity here is that inside `get_dim_groups` of DeviceMesh, it still uses the [find_pg_by_tags_and_rank](https://github.com/pytorch/pytorch/blob/main/torch/distributed/_device_mesh.py#L348) to search for a PG in the c10d's global map in Python, wondering if you have ideas how to do this PG resolution in the future?

Reply:
Currently `_dim_group_infos` tracks `tag` and `rank_set`. I planned to add `group_name` in addition (and eventually remove `tag` and `rank_set`).